### PR TITLE
Use apt layer to manage configurable sources

### LINKED ISSUE #2: Using an external repository breaks enterprise charms
Suggest to have configurable source and key with defaults to nodesource repository so that deployments with limited egress/ingress filtering can still use nodejs charms without having to resort to archive only

### PR DESCRIPTION
**Pull request by marcoceppi**
This pull request is a bit, heavy handed. I'm interested in feedback on the approach esp since it guts quite a bit of the current logic. Attempts to fix #2

### CODE REVIEW

**Comment by adam-stokes** [comment]
@marcoceppi so I actually really like this approach, once you find out about the apt layer symlinking reactive we should get this merged in.

**Comment by marcoceppi** [comment]
@battlemidget the apt layer has been updated and so has this merge request

File: lib/charms/layer/nodejs.py [hunk 1]
```diff
@@ -1,8 +1,12 @@
 import os
 import sys
+
 from shell import shell
+from charms import layer
```
Comment (adam-stokes) [inline lib/charms/layer/nodejs.py]: Just curious what this import is for?

Comment (adam-stokes) [inline lib/charms/layer/nodejs.py]: nm i saw the location of the file being moved, so that's the preferred way now is to have everything in ./charms/layer

**Comment by adam-stokes** [comment]
These changes look good to me!, also added you as a collaborator @marcoceppi

File: lib/charms/layer/nodejs.py (hunk 1 above)
Comment (marcoceppi) [inline lib/charms/layer/nodejs.py]: It is, but layer also has the options for parsing stuff from layer.yaml, but I refactored that code out and need to remove this import now

**Pull request merged**